pkg/kubectl/cmd/annotation.go

```diff
@@ -0,0 +1,288 @@
+				return err
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+// parseAnnotations retrieves new and remove annotations from annotation args
+func parseAnnotations(annotationArgs []string) (newAnnotations map[string]string, removeAnnotations []string) {
+	newAnnotations = make(map[string]string)
+	removeAnnotations = []string{}
+	for _, annotationArg := range annotationArgs {
+		if strings.Index(annotationArg, "=") != -1 {
+			parts := strings.SplitN(annotationArg, "=", 2)
+			if len(parts) == 2 && len(parts[1]) > 0 {
+				newAnnotations[parts[0]] = parts[1]
+			}
+		} else if strings.HasSuffix(annotationArg, "-") {
+			removeAnnotations = append(removeAnnotations, annotationArg[:len(annotationArg)-1])
+		}
+	}
+	return
+}
+
+// validateAnnotations checks the format of annotation args and checks removed annotations aren't in the new annotations map
+func validateAnnotations(annotationArgs, removeAnnotations []string, newAnnotations map[string]string) error {
+	var invalidBuf, modifyRemoveBuf bytes.Buffer
+	for _, annotationArg := range annotationArgs {
+		if strings.Index(annotationArg, "=") != -1 {
+			parts := strings.SplitN(annotationArg, "=", 2)
+			if len(parts) != 2 || len(parts[1]) == 0 {
+				if invalidBuf.Len() > 0 {
+					invalidBuf.WriteString(", ")
+				}
+				invalidBuf.WriteString(fmt.Sprintf(annotationArg))
+			}
+		} else if !strings.HasSuffix(annotationArg, "-") {
+			if invalidBuf.Len() > 0 {
+				invalidBuf.WriteString(", ")
+			}
+			invalidBuf.WriteString(fmt.Sprintf(annotationArg))
+		}
+	}
+	if invalidBuf.Len() > 0 {
+		return fmt.Errorf("invalid annotation format: %s", invalidBuf.String())
+	}
+
+	for _, removeAnnotation := range removeAnnotations {
+		if _, found := newAnnotations[removeAnnotation]; found {
+			if modifyRemoveBuf.Len() > 0 {
+				modifyRemoveBuf.WriteString(", ")
+			}
+			modifyRemoveBuf.WriteString(fmt.Sprintf(removeAnnotation))
+		}
+	}
+	if modifyRemoveBuf.Len() > 0 {
+		return fmt.Errorf("can not both modify and remove the following annotation(s) in the same command: %s", modifyRemoveBuf.String())
+	}
+
+	return nil
+}
+
+// validateNoAnnotationOverwrites validates that when overwrite is false, to-be-updated annotations don't exist in the object annotation map (yet)
+func validateNoAnnotationOverwrites(meta *api.ObjectMeta, annotations map[string]string) error {
+	var buf bytes.Buffer
+	for key := range annotations {
+		if value, found := meta.Annotations[key]; found {
+			if buf.Len() > 0 {
+				buf.WriteString("; ")
+			}
+			buf.WriteString(fmt.Sprintf("'%s' already has a value (%s)", key, value))
+		}
+	}
+	if buf.Len() > 0 {
+		return fmt.Errorf("--overwrite is false but found the following declared annotation(s): %s", buf.String())
+	}
+	return nil
+}
+
+// updateAnnotations updates annotations of obj
+func updateAnnotations(obj runtime.Object, overwrite bool, resourceVersion string, annotations map[string]string, remove []string) error {
```

Comment:
if you had your `AnnotationOptions` as a method receiver, most of the args would fall out.